dds/DCPS/transport/rtps_udp/RtpsUdpDataLink.cpp

```diff
@@ -3454,7 +3473,73 @@ void RtpsUdpDataLink::durability_resend(TransportQueueElement* element,
   const TqeVector::iterator end = to_send.end();
   for (TqeVector::iterator i = to_send.begin(); i != end; ++i) {
     if (fragments.empty() || include_fragment(**i, fragments, lastFragment)) {
-      send_strategy()->send_rtps_control(*const_cast<ACE_Message_Block*>((*i)->msg()), addrs);
+      RTPS::Message message;
+      send_strategy()->send_rtps_control(message, *const_cast<ACE_Message_Block*>((*i)->msg()), addrs);
+      if (transport_debug.log_messages) {
```

Comment:
Could this whole block move into RTPS library's Logging.cpp?  Or does it need something in the transport library?